analyzers/src/SonarAnalyzer.Common/Rules/BackslashShouldBeAvoidedInAspNetRoutesBase.cs

```diff
@@ -56,10 +56,21 @@ protected void Check(SonarSyntaxNodeReportingContext c)
         && model.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol
         && Language.MethodParameterLookup(invocation, methodSymbol) is { } parameterLookup
         && parameterLookup.TryGetSymbol(node, out var parameter)
-        && methodSymbol.ContainingType.IsAny(KnownType.RouteAttributes) && methodSymbol.IsConstructor() && parameter.Name == "template";
+        && (HasStringSyntaxAttributeOfTypeRoute(parameter) || IsRouteTemplateBeforeAspNet6(parameter, methodSymbol));
+
+    private static bool HasStringSyntaxAttributeOfTypeRoute(IParameterSymbol parameter) =>
+        parameter.GetAttributes(KnownType.System_Diagnostics_CodeAnalysis_StringSyntaxAttribute).FirstOrDefault() is { } syntaxAttribute
+        && syntaxAttribute.TryGetAttributeValue<string>("syntax", out var syntaxParameter)
+        && string.Equals(syntaxParameter, "Route", StringComparison.Ordinal);
+
+    private static bool IsRouteTemplateBeforeAspNet6(IParameterSymbol parameter, IMethodSymbol method) =>
+        (method.ContainingType.IsAny(KnownType.RouteAttributes)
+            || method.ContainingType.DerivesFrom(KnownType.Microsoft_AspNetCore_Mvc_Routing_HttpMethodAttribute))
```

Comment:
Because route attributes are not in the same hierarchy. When introducing ASP.NET Core, Microsoft decided to change the namespace, so attributes are in no hierarchical relationship with each other.

On the other hand, all HTTP methods-related attributes in ASP.NET Core derive from the same abstract attribute [`HttpMethodAttribute`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.routing.httpmethodattribute?view=aspnetcore-8.0), and that allows us to just check that `method.ContainingType` derives from this abstract attribute.

Notice how:
- `HttpMethodAttribute` doesn't exist in ASP.NET 4.x
-  in any case, we don't want to include `method.ContainingType.DerivesFrom(KnownType.System.Web.Mvc.ActionMethodSelectorAttribute)` (base attribute of `HttpGetAttribute` and similar in ASP.NET 4.x), because in ASP.NET 4.x HTTP methods-related attribute don't allow to specify a route attribute.

I can add a small comment `// Route templates cannot be specified in HttpXAttribute in ASP.NET 4.x`, if you consider that it helps.

Reply:
No, it's good. I was just wondering, thanks a lot for the answer!